dist/functions.setup-appliance.sh

```diff
@@ -143,12 +143,19 @@ function get_hostname {
   else
     TIMEOUT=30
     while [ -z "$FQHOSTNAME" -o "$FQHOSTNAME" = "localhost" ];do
-      FQHOSTNAME=`hostname -f 2>/dev/null`
+      # Try to get the FQHN via hostname
+      HN=`hostname -f 2>/dev/null`
+      if [[ $HN =~ ^[A-Za-z0-9-]+(\.[A-Za-z0-9-]+)+$ ]];then
+        FQHOSTNAME=$HN
+      fi
       TIMEOUT=$(($TIMEOUT-1))
       [ "$TIMEOUT" -le 0 ] && break
       echo "Waiting for FQHOSTNAME ($TIMEOUT)"
       sleep 1
     done
+    if [ -z "$FQHOSTNAME" -o "$FQHOSTNAME" = "localhost" ];then
+      FQHOSTNAME=`hostnamectl --static 2>/dev/null`
```

Comment:
Are you sure that you want to accept the return value of `hostnamectl --static` even if it does not match a FQHN?
And I do not see a fallback to /etc/hostname
Might be useful  if hostnamectl is not installed.

Reply:
i remove `cat /etc/hostname` was because i found hostnamectl is reading it and it's systemd's func, but systemd may not be included in container image or not using systemd as init after you remind, so you're right